onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1375,6 +1375,21 @@ ONNX_MS_OPERATOR_SET_SCHEMA(Sampling, 1,
                                   GreedySearchShapeInference(ctx);
                                 }));
 
+ONNX_MS_OPERATOR_SET_SCHEMA(MoEBlock, 1,
+                            OpSchema()
+                                .SetDoc("Mixture of experts.")
+                                .Attr("activation_type", "Activation function to use", AttributeProto::STRING, std::string("relu"))
+                                .Attr("k", "Number of top experts to select from expert pool", AttributeProto::INT, static_cast<int64_t>(1))
+                                .Input(0, "input", "2D input tensor with shape (num_rows, hidden_size)", "T")
+                                .Input(1, "gated_output", "2D input tensor with shape (num_rows, num_experts)", "T")
+                                .Input(2, "fc1_experts_weights", "3D input tensor with shape (num_experts, hidden_size, inter_size)", "T")
+                                .Input(3, "fc2_experts_weights", "3D input tensor with shape (num_experts, inter_size, hidden_size)", "T")
```

Comment:
How often MoE has two FC layers? I saw some paper uses only one FC layer for MoE.

Reply:
I don't know. until now the models I see are the 2-FC version. we can support the single FC case when there is a requirement